### DISCUSSION TITLE
How to map the size of a page to real-life dimensions?

### DISCUSSION THREAD

**Discussion by arun-mani-j** [pymupdf/PyMuPDF#2163] [Looking for help] [answered]
Hii.
This could be more into pixels logic than MuPDF, but I'm asking here since this is the closest place I'm aware of.

Using `Page.rect.width` and `Page.rect.height`, one could get the width and height of a page. My aim is to display the page in the same dimensions if it was printed. 

For example, let us assume `page.rect.width, page.rect.height` returns `(595, 842)` pixels which means A4 size. The [size of an A4 sheet](https://a-size.com/a4-paper-size/) is 8.3 x 11.7 inches. From the [W3C CSS Values](https://www.w3.org/TR/css3-values/#absolute-lengths), 1 inch equals 96 pixels. 

So `(8.3, 11.7)` x 96 equals `(796.8, 1123.1)` pixels. Therefore, to display an A4 page, I have to scale it to `796.8, 1123.1` pixels, so that it matches the real dimensions if it was printed out.

Here I knew that the page is A4 and was able to perform the calculations manually. But what about arbitrary lengths, where the size is unknown or is not a standard paper size? How do PDF viewers like GNOME's Evince or KDE's Okular do it?

Any kind of suggestions are highly appreciated. Thanks for reading!

**Comment by JorjMcKie** [comment, marked as answer]
That's an easy answer:
1. There is `fitz.paper_sizes` a dictionary with key of paper format, like a4 or letter, and values of width and height - in portrait version.
2. Landscape version simply is width / height exchanged.
3. The pair of numbers are integers in point, where 72 points are one inch.

Example for format "Letter": (612, 792). `612 / 72 = 8.5 inches = 21.59 cm` width, `792 / 72 = 11 inches = 27,94 cm` height.

**Reply by arun-mani-j** [reply]
Thanks for the answer!

This works only for the standard sizes right? What if the size is custom? I searched for "get paper size of PDF in Linux" and got this [answer which uses `mutool`](https://unix.stackexchange.com/a/682949).

Here they say that `mutool` returns page sizes in `pts` which is 1/72 of an inch. [W3C CSS Values](https://www.w3.org/TR/css3-values/#absolute-lengths) defines 1 in as 96 px.

So if I multiply the values returned by `page.rect.width` and `page.rect.height` by `96/72`, I should get the real dimensions right?

Taking the well known example of A4 size.
```python
>>> w, h = page.rect.width, page.rect.height
>>> w, h
(595.3499755859375, 841.9500122070312)
>>> f = 96 / 72
>>> nw, nh = w * f, h * f
>>> nw, nh
(793.7999674479166, 1122.6000162760415)
```
which is pretty close to the actual size `(8.3 * 96, 11.7 * 96) = (796.8, 1123.2)` ...

So what I'm doing is right or is there a better way which works good for non-standard sizes?
Again thanks for the help :)

**Reply by JorjMcKie** [reply]
`page.rect`, `CropBox`, `MediaBox` always means points - to be treated as described above.
A different thing is the DPI (like 96 in your example). This value has nothing to do with the size of the paper of the printout, but with the resolution (color pixels per inch in each direction).

**Reply by arun-mani-j** [reply]
Oh thanks again. Also thanks for PyMuPDF :)
